### PR TITLE
Misc fixes to docs, links, error messages, formatting, etc.

### LINKED ISSUE #1142: Examples database keeps changing even when no failures are found
Is this intended? It makes it hard to version-control the database.

### LINKED ISSUE #1219: Various docs tweaks I've been meaning to do
- [x] Give each version in the changelog a persistent link reference (so that links keep working after the next update)
- [x] Update the page on projects using Hypothesis to point to https://libraries.io/pypi/hypothesis as well as the explicit list - it already links to 700+ repos depending on Hypothesis.
- [x] Update the page on projects *extending* Hypothesis based on libraries.io and from a search of pypi.org (also link to the relevant searches so new extensions get some exposure too)
- [x] Note that `@seed(...)` is designed for reproducing failures observed on a remote system (eg CI); and that `@settings(derandomize=True)` is more suitable if you want the same cases to run every time (for a particular version of Hypothesis).
- [x] Docs to resolve #1142
- [x] Pull the non-v4-specific improvements to the docs out of #1156 
- [x] Correct the module referred to by datetimes() (should be `hypothesis.extra.pytz.timezones`; missing `.pytz`)

Checked items in [my WIP branch](https://github.com/HypothesisWorks/hypothesis-python/compare/master...Zac-HD:misc-docs); I'll open a pull when it's ready.

### LINKED ISSUE #1225: test_minmax_magnitude_equal does not handle overflow to infinity
```python-traceback
_________________________ test_minmax_magnitude_equal _________________________
Traceback (most recent call last):
  File "C:\projects\hypothesis\hypothesis-python\tests\cover\test_complex_numbers.py", line 119, in test_minmax_magnitude_equal
    def test_minmax_magnitude_equal(data, mag):
  File "C:\projects\hypothesis\hypothesis-python\tests\cover\test_complex_numbers.py", line 119, in test_minmax_magnitude_equal
    def test_minmax_magnitude_equal(data, mag):
  File "C:\projects\hypothesis\hypothesis-python\tests\cover\test_complex_numbers.py", line 122, in test_minmax_magnitude_equal
    assert math.isclose(abs(val), mag)
AssertionError: assert False
 +  where False = <built-in function isclose>(inf, 1.7976931348623155e+308)
 +    where <built-in function isclose> = math.isclose
 +    and   inf = abs((inf+3.99168061906944e+292j))
--------------------------------- Hypothesis ----------------------------------
Falsifying example: test_minmax_magnitude_equal(data=data(...), mag=1.7976931348623155e+308)
Draw 1: (inf+3.99168061906944e+292j)
```

Via [Appveyor](https://ci.appveyor.com/project/DRMacIver/hypothesis/build/1.0.3/job/2o1tp60dfw3w4708).  Adding `assume(abs(val) < float('inf'))` would be a sufficient fix for this.

(note: I've edited the traceback as suggested in #848, which makes the problem much more obvious)

### PR DESCRIPTION
**Pull request by Zac-HD**
Fixes #1225, closes #1219, closes #1142, steals parts of #1156.

### CODE REVIEW

**Comment by Zac-HD** [comment]
Ready to merge!

**Comment by Zac-HD** [comment]
Ping @DRMacIver; this includes some build fixes for 32bit Windows as well as the headline docs so it would be nice to merge soon :smile:

File: hypothesis-python/docs/usage.rst [hunk 1]
```diff
@@ -2,10 +2,14 @@
 Open Source Projects using Hypothesis
 =====================================
 
-The following is a non-exhaustive list of open source projects I know are using Hypothesis. If you're aware of
-any others please add them to the list! The only inclusion criterion right now is that if it's a Python library
+The following is a non-exhaustive list of open source projects I know are
```
Comment (DRMacIver) [inline hypothesis-python/docs/usage.rst]: Maybe switch "I" to "we"?

I wonder if this list is even useful given libraries.io?

File: hypothesis-python/docs/strategies.rst [hunk 2]
```diff
@@ -2,22 +2,77 @@
...
+* :pypi:`hypothesis-mongoengine` - infer strategies from a :pypi:`mongoengine` model.
+* :pypi:`hypothesis-pb` - infer strategies from `Protocol Buffer
+  <https://developers.google.com/protocol-buffers/>`_ schemas.
+
+
+-----------------
+Other Cool Things
+-----------------
+
+:pypi:`swagger-conformance` is powered by Hypothesis and :pypi:`pyswagger`.
```
Comment (DRMacIver) [inline hypothesis-python/docs/strategies.rst]: Huh. Interesting.

File: hypothesis-python/docs/usage.rst (hunk 1 above)
Comment (Zac-HD) [reply hypothesis-python/docs/usage.rst]: I'm basically just going for incremental fixes here, with a larger overhaul on the horizon.  Specifically I'm working on #1193 this weekend (hopefully including auto-gen API reference) and probably a short tutorial by late May for pycon.sg, which covers the easy half of #774.

**Pull request merged**